doc: nrf: add nrf_peripherals chapter (#27437)
nrfconnect/sdk-nrf: doc/nrf/peripherals_drivers.rst

.. _peripherals_drivers:

Peripherals drivers
###################

> Nordic Semiconductor SoCs contain a variety of hardware peripherals that can be used for many purposes in context of |NCS| applications.

Reviewer: ```suggestion
Nordic Semiconductor SoCs contain a variety of hardware peripherals that can be used for many purposes in the context of |NCS| applications.
```

Some of them can be utilized using `Zephyr's device driver API`_, while others are accessible only through `nrfx`_.

> There are also various subsystems or libraries available, that may access hardware peripheral underneath.

Reviewer: ```suggestion
Various subsystems and libraries are also available that may access the underlying hardware peripherals.
```

> Here you can find documentation for using supported hardware peripherals in the |NCS|.

Reviewer: ```suggestion
The following pages describe how to use the supported hardware peripherals in the |NCS|.
```
